### PR TITLE
feat(23.10) add slice for crun and uidmap

### PR DESCRIPTION
**Pull request by endersonmaia**
# Proposed changes

### Forward porting

- https://github.com/canonical/chisel-releases/pull/222
- https://github.com/canonical/chisel-releases/pull/223
- https://github.com/canonical/chisel-releases/pull/225

## Testing

```bash
❯ chisel cut \
    --release ./ \
    --root rootfs/ \
    crun_bins uidmap_bins
2024/04/20 12:04:16 Processing ./ release...
2024/04/20 12:04:16 Selecting slices...
2024/04/20 12:04:16 Fetching ubuntu 23.10 mantic suite details...
2024/04/20 12:04:17 Release date: Mon, 16 Oct 2023 21:12:12 UTC
2024/04/20 12:04:17 Fetching index for ubuntu 23.10 mantic main component...
2024/04/20 12:04:17 Fetching index for ubuntu 23.10 mantic universe component...
2024/04/20 12:04:17 Fetching ubuntu 23.10 mantic-security suite details...
2024/04/20 12:04:17 Release date: Fri, 19 Apr 2024 13:01:58 UTC
2024/04/20 12:04:17 Fetching index for ubuntu 23.10 mantic-security main component...
2024/04/20 12:04:17 Fetching index for ubuntu 23.10 mantic-security universe component...
2024/04/20 12:04:17 Fetching ubuntu 23.10 mantic-updates suite details...
2024/04/20 12:04:18 Release date: Sat, 20 Apr 2024  7:55:11 UTC
2024/04/20 12:04:18 Fetching index for ubuntu 23.10 mantic-updates main component...
2024/04/20 12:04:18 Fetching index for ubuntu 23.10 mantic-updates universe component...
2024/04/20 12:04:18 Fetching pool/main/g/glibc/libc6_2.38-1ubuntu6.2_amd64.deb...
2024/04/20 12:04:18 Fetching pool/main/libc/libcap2/libcap2_2.66-4ubuntu1_amd64.deb...
2024/04/20 12:04:18 Fetching pool/main/libs/libseccomp/libseccomp2_2.5.4-1ubuntu3_amd64.deb...
2024/04/20 12:04:18 Fetching pool/main/libg/libgpg-error/libgpg-error0_1.47-2_amd64.deb...
2024/04/20 12:04:18 Fetching pool/main/libg/libgcrypt20/libgcrypt20_1.10.2-3ubuntu1_amd64.deb...
2024/04/20 12:04:18 Fetching pool/main/l/lz4/liblz4-1_1.9.4-1_amd64.deb...
2024/04/20 12:04:18 Fetching pool/main/x/xz-utils/liblzma5_5.4.1-0.2_amd64.deb...
2024/04/20 12:04:18 Fetching pool/main/libz/libzstd/libzstd1_1.5.5+dfsg2-1ubuntu2_amd64.deb...
2024/04/20 12:04:18 Fetching pool/main/s/systemd/libsystemd0_253.5-1ubuntu6.1_amd64.deb...
2024/04/20 12:04:18 Fetching pool/main/y/yajl/libyajl2_2.1.0-5_amd64.deb...
2024/04/20 12:04:18 Fetching pool/universe/c/crun/crun_1.8.5-1_amd64.deb...
2024/04/20 12:04:18 Fetching pool/main/a/audit/libaudit-common_3.1.1-1_all.deb...
2024/04/20 12:04:18 Fetching pool/main/libc/libcap-ng/libcap-ng0_0.8.3-1build2_amd64.deb...
2024/04/20 12:04:18 Fetching pool/main/a/audit/libaudit1_3.1.1-1_amd64.deb...
2024/04/20 12:04:18 Fetching pool/main/p/pcre2/libpcre2-8-0_10.42-4_amd64.deb...
2024/04/20 12:04:18 Fetching pool/main/libs/libselinux/libselinux1_3.5-1_amd64.deb...
2024/04/20 12:04:18 Fetching pool/main/s/shadow/uidmap_4.13+dfsg1-1ubuntu1.1_amd64.deb...
2024/04/20 12:04:18 Extracting files from package "libc6"...
2024/04/20 12:04:18 Extracting files from package "libcap2"...
2024/04/20 12:04:18 Extracting files from package "libseccomp2"...
2024/04/20 12:04:18 Extracting files from package "libgpg-error0"...
2024/04/20 12:04:18 Extracting files from package "libgcrypt20"...
2024/04/20 12:04:18 Extracting files from package "liblz4-1"...
2024/04/20 12:04:18 Extracting files from package "liblzma5"...
2024/04/20 12:04:18 Extracting files from package "libzstd1"...
2024/04/20 12:04:18 Extracting files from package "libsystemd0"...
2024/04/20 12:04:18 Extracting files from package "libyajl2"...
2024/04/20 12:04:18 Extracting files from package "crun"...
2024/04/20 12:04:18 Extracting files from package "libaudit-common"...
2024/04/20 12:04:18 Extracting files from package "libcap-ng0"...
2024/04/20 12:04:18 Extracting files from package "libaudit1"...
2024/04/20 12:04:18 Extracting files from package "libpcre2-8-0"...
2024/04/20 12:04:18 Extracting files from package "libselinux1"...
2024/04/20 12:04:18 Extracting files from package "uidmap"...
```

## Checklist

* [x] I have read the [contributing guidelines](
https://github.com/canonical/chisel-releases/blob/main/CONTRIBUTING.md)
* [x] I have already submitted the [CLA form](
https://ubuntu.com/legal/contributors/agreement)

### CODE REVIEW

**Review by zhijie-yang — CHANGES_REQUESTED** [review]
Looks good. There are some issues regarding the linting (see workflow runs) to be fixed, and it is suggested to add the copyright files explicitly in the slice definitions.

**Review by zhijie-yang — APPROVED** [review]
Looks all good to me! Thanks!

File: slices/libcap-ng0.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+essential:
+  - libcap-ng0_copyright
+
+slices:
+  libs:
+    essential:
+      - libc6_libs
+    contents:
+      /lib/*-linux-*/libcap-ng.so.0*:
```
Comment (linostar) [inline slices/libcap-ng0.yaml:11]: missing `/lib/x86_64-linux-gnu/libdrop_ambient.so.0*:`

File: slices/libsubid4.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+esential:
+  - libsubid4_copyright
+
+slices:
+  libs:
+    essential:
+      - libaudit1_libs
+      - libc6_libs
+      - libselinux1_libs
```
Comment (linostar) [inline slices/libsubid4.yaml:11]: missing dependencies:
- libcrypt1
- libpam0g
- libsemanage2

File: slices/uidmap.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  - uidmap_copyright
+
+slices:
+  bins:
+    essential:
+      - libaudit1_libs
+      - libc6_libs
+      - libselinux1_libs
+      - libsubid4_libs
+    contents:
```
Comment (linostar) [inline slices/uidmap.yaml:13]: missing `/usr/bin/getsubids:`

**Review by linostar — CHANGES_REQUESTED** [review]

File: slices/libsubid4.yaml (hunk 2 above)
Comment (endersonmaia) [reply slices/libsubid4.yaml:11]: Here I get only these: `Depends: libaudit1 (>= 1:2.2.1), libc6 (>= 2.38), libselinux1 (>= 3.1~)`

Comment (linostar) [reply slices/libsubid4.yaml:11]: Those extra libraries are needed for armhf, i386 and riscv64 architectures.

**Review by linostar — CHANGES_REQUESTED** [review]
Please see my comments https://github.com/canonical/chisel-releases/pull/225#pullrequestreview-2069161775 and https://github.com/canonical/chisel-releases/pull/225#discussion_r1608658239 in PR #225 and address them here as well.

**Comment by endersonmaia** [comment]
> Please see my comments [#225 (review)](https://github.com/canonical/chisel-releases/pull/225#pullrequestreview-2069161775) and [#225 (comment)](https://github.com/canonical/chisel-releases/pull/225#discussion_r1608658239) in PR #225 and address them here as well.

The file is already there https://github.com/endersonmaia/chisel-releases/blob/crun-to-23.10/slices/libcrypt1.yaml

**Review by linostar — APPROVED** [review]
LGTM. Thanks!

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
ty for the proposal. 

Apart from the linting, I'd suggest also re-working the `debconf` slice definitions. Especially the `bins` slice, as it contains much more than just binaries. Those modules deserve a separate slice(s). There are two preferred approaches here: group the modules into functionality, and create slices for each one of those (like we did with the python and openjdk slices); the other approach is to create a generic `modules` slices where all of those modules fallback into.

**Comment by endersonmaia** [comment]
I was experimenting debconf, and although it doesn't explicitly show  `perl` dependency (it suggests), it doesn't work without it, and I see no `perl` slice available.

```
# apt depends debconf
debconf
  Conflicts: <debconf-tiny>
  Conflicts: <whiptail-utf8> (<= 0.50.17-13)
  Recommends: apt-utils
  Recommends: debconf-i18n
  Suggests: debconf-doc
  Suggests: debconf-kde-helper
  Suggests: debconf-utils
  Suggests: libgtk3-perl
  Suggests: libnet-ldap-perl
  Suggests: libterm-readline-gnu-perl
  Suggests: perl
 |Suggests: whiptail
  Suggests: dialog
  Replaces: <debconf-tiny>
```

I didn't get the point of installing its bins if I can't use it.

```shell
# head /usr/bin/debconf -n 1
#!/usr/bin/perl
```

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
right, so this is a fundamental question for this PR: if `debconf` doesn't work without `perl`, how did these slices work for your use case?
_Please note that in some cases `debconf` is a dependency of a pkg because it is used during its `postinst`. If your `crun`+`uidmap` use case works well with the current slices and mutation scripts, then I'd consider leaving the `debconf` slices out of this PR -> no need to introduce something we're sure about and that hasn't been thoroughly tested_

Regardless of that dependency, if we want to keep the `debconf` SDF, it still needs to be reworked according to the previous comment.

**Comment by cjdcordeiro** [comment]
let's close this since the release is EOL. #223 is going in soon

**Pull request closed without merging**